tests/login.go

```diff
@@ -45,30 +45,29 @@ func LoggedInCirrosExpecter(vmi *v1.VirtualMachineInstance) (expect.Expecter, er
 	}
 
 	b := append([]expect.Batcher{
-		&expect.BSnd{S: "\n"},
 		&expect.BSnd{S: "\n"},
 		&expect.BExp{R: "login as 'cirros' user. default password: 'gocubsgo'. use 'sudo' for root."},
 		&expect.BSnd{S: "\n"},
 		&expect.BExp{R: hostName + " login:"},
 		&expect.BSnd{S: "cirros\n"},
 		&expect.BExp{R: "Password:"},
 		&expect.BSnd{S: "gocubsgo\n"},
-		&expect.BExp{R: "\\$"}})
-	resp, err := expecter.ExpectBatch(b, 180*time.Second)
+		&expect.BExp{R: PromptExpression}})
+	resp, err := ExpectBatchWithValidatedSend(expecter, b, 180*time.Second)
```

Comment:
I wouldn't change the login methods to use the `ExpectBatchWithValidatedSend`.
The login batch starts with searching a text that is already in the expecter buffer. Since `ExpectBatchWithValidatedSend` assuming the expected expression is the output of the sent command, it is not suitable in this case.